### PR TITLE
De-flake `BuildKeepsRunningWhenFaultySubTasksTest`

### PR DESCRIPTION
**Pull request by basil**
`BuildKeepsRunningWhenFaultySubTasksTest` flaked with:

```
01:21:46  [ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 12.488 s <<< FAILURE! - in hudson.model.queue.BuildKeepsRunningWhenFaultySubTasksTest
01:21:46  [ERROR] hudson.model.queue.BuildKeepsRunningWhenFaultySubTasksTest.buildFinishesWhenSubTaskFails  Time elapsed: 12.47 s  <<< ERROR!
01:21:46  java.util.concurrent.ExecutionException: java.lang.InterruptedException
01:21:46  	at hudson.remoting.AsyncFutureImpl.get(AsyncFutureImpl.java:80)
01:21:46  	at hudson.remoting.AsyncFutureImpl.get(AsyncFutureImpl.java:108)
01:21:46  	at hudson.model.queue.BuildKeepsRunningWhenFaultySubTasksTest.buildFinishesWhenSubTaskFails(BuildKeepsRunningWhenFaultySubTasksTest.java:37)
01:21:46  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
01:21:46  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
01:21:46  	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
01:21:46  	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
01:21:46  	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
01:21:46  	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
01:21:46  	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
01:21:46  	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
01:21:46  	at org.jvnet.hudson.test.JenkinsRule$1.evaluate(JenkinsRule.java:613)
01:21:46  	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:299)
01:21:46  	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:293)
01:21:46  	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
01:21:46  	at java.base/java.lang.Thread.run(Thread.java:829)
01:21:46  Caused by: java.lang.InterruptedException
01:21:46  	at hudson.model.queue.Latch.check(Latch.java:104)
01:21:46  	at hudson.model.queue.Latch.synchronize(Latch.java:83)
01:21:46  	at hudson.model.queue.WorkUnitContext.synchronizeStart(WorkUnitContext.java:132)
01:21:46  	at hudson.model.Executor.run(Executor.java:408)
01:21:46  Caused by: hudson.AbortException
01:21:46  	at hudson.model.queue.Latch.abort(Latch.java:58)
01:21:46  	at hudson.model.queue.WorkUnitContext.abort(WorkUnitContext.java:204)
01:21:46  	at hudson.model.Executor.finish1(Executor.java:492)
01:21:46  	at hudson.model.Executor.run(Executor.java:471)
01:21:46  Caused by: java.lang.ArrayIndexOutOfBoundsException: My unexpected exception
01:21:46  	at hudson.model.queue.BuildKeepsRunningWhenFaultySubTasksTest$FailingSubTaskContributor$1$1.run(BuildKeepsRunningWhenFaultySubTasksTest.java:58)
01:21:46  	at hudson.model.ResourceController.execute(ResourceController.java:107)
01:21:46  	at hudson.model.Executor.run(Executor.java:449)
```

The test builds a job and expects it to succeed despite logging a warning due to a failing `SubTaskContributor`. Yet the test does not assert that the warning was logged, nor does it wait a sufficient amount of time for the job to complete: rather, it waits an arbitrary amount of time of five seconds. As [Joshua Clulow wrote](https://twitter.com/jmclulow/status/879362528911872000):

> Timeouts, timeouts: always wrong! Some too short and some too long.

The fix is to remove the arbitrary five-second timeout in favor of the global 180-second test timeout. While I was here I also improved stability on Windows by making sure we always wait for the build to complete. And while I was here I also improved the test so that it asserts that the expected warning was indeed logged.

### Proposed changelog entries

N/A

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change) and are in the imperative mood. [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  - Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadoc, as appropriate.
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")` if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content-Security-Policy directives (see [documentation on jenkins.io](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`:

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are accurate, human-readable, and in the imperative mood
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/7168"></a>

### CODE REVIEW

**Review by NotMyFault — APPROVED** [review]

**Review by timja — APPROVED** [review]

**Comment by NotMyFault** [comment]
This PR is now ready for merge. We will merge it after ~24 hours if there is no negative feedback.
Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process.
Thanks!

**Pull request merged**
